app/src/test/java/com/example/trainingintive/dog_images_feature/presentation/DogImageViewModelTest.kt

```diff
@@ -36,20 +39,20 @@ class DogImageViewModelTest : RulesForTests() {
 
     @Test
     fun `after successfully download, dogimage should be inserted into local database`() {
-        DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
         verify { repository.insert(sampleDog) }
     }
 
     @Test
-    fun `if repository returns error livedata should contain message`() {
-        val errorMessage = "Unable to resolve host \"dog.ceo\": No address associated with hostname"
-        every { repository.getDogImageUrl() } returns Single.error(Exception(errorMessage))
+    fun `if repository returns error vm should send error event`() {
+        val exception: Exception = mockk()
+        mockkStatic(exception::toErrorTextId)
+        every { repository.getDogImageUrl() } returns Single.error(exception)
 
-        val tested = DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
-        tested.dog.observeForever { result ->
-            result shouldBeEqualTo errorMessage
-        }
+        verify { exception.toErrorTextId() }
```

Comment:
Minor: It should be tested in another test function for readability :D try to test one thing in one test when it is possible